### PR TITLE
Fix instances count logging

### LINKED ISSUE #24737: ‘Instances’ column in training progress bar shows image batch size instead of instance count
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

In detection training, the progress bar column is still labeled `Instances`, but the value being printed is the image batch size of the current rank, not the number of object instances (It used to be but was changed during commit 93e0555251bd7eef5ac8fdf367200d32a5967053).

In current upstream `main` at commit `3ca0b4fc373c01522da1a6ec25710516ae21beb2`:

```python
# ultralytics/models/yolo/detect/train.py
"Instances",
```

But the actual value logged in the progress bar is:

```python
# ultralytics/engine/trainer.py
batch["img"].shape[0],  # batch size, i.e. 8
```

This is misleading, especially in DDP training. For example, with global `batch=1024` over 16 DDP processes, the progress bar always shows `Instances=64`, but `64` is the per-rank image batch size, not the number of ground-truth object instances.

### Environment

This issue is not hardware-specific. It is reproducible by source inspection on upstream `main`:

```text
Ultralytics YOLO commit: 3ca0b4fc373c01522da1a6ec25710516ae21beb2
Component: Train
Task: detect
Mode: train
```

The mismatch exists in the upstream source:

```python
# Header:
# ultralytics/models/yolo/detect/train.py
"Instances"

# Logged value:
# ultralytics/engine/trainer.py
batch["img"].shape[0]
```

Observed during distributed detection training, but the bug is independent of OS, CUDA/NPU, or DDP. DDP only makes it more obvious because the displayed value becomes the per-rank image batch size.

### Minimal Reproducible Example

```python
from ultralytics import YOLO

model = YOLO("yolo11n.yaml")

model.train(
    data="coco8.yaml",
    imgsz=640,
    epochs=1,
    batch=4,
)
```

### Additional

This appears to have been introduced in:

```text
93e0555251bd7eef5ac8fdf367200d32a5967053
`ultralytics 8.4.52` YOLO Semantic Segmentation models (#24518)
```

Relevant diff:

```diff
- batch["cls"].shape[0],  # batch size, i.e. 8
+ batch["img"].shape[0],  # batch size, i.e. 8
```

The value changed from label/object count to image batch size, but the detection trainer progress header still says `Instances`.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Y-T-G**
Closes #24737 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🛠️ This PR fixes training progress reporting so the logged count reflects the number of classification samples or detection instances more accurately, instead of always showing the image batch size.

### 📊 Key Changes
- Updated the training log output in `ultralytics/engine/trainer.py`.
- Changed the reported count from `batch["img"].shape[0]` to `batch.get("cls", batch["img"]).shape[0]`.
- This means:
  - For classification tasks, the trainer now uses the `cls` tensor length when available.
  - For other tasks, it safely falls back to the image batch size as before.
- The displayed metric label comment was also adjusted from “batch size” to “no. of instances” to better match what is being reported. 🧾

### 🎯 Purpose & Impact
- Improves the accuracy of training progress logs, especially for tasks where image count is not the best representation of what is being processed. ✅
- Makes console output clearer and more consistent across different training modes. 👀
- Helps users better interpret training behavior without changing model training logic or results. 🚀
- Low-risk change: this only affects logging/display, not the actual optimization or dataset handling. 🔒

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you shortly. Please review the checklist below to help ensure a smooth integration 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The change looks clean and well-scoped. Updating the progress display to prefer `cls` count aligns the logged value with instance-based tasks, and I don't see any clear bug or maintainability issue in the diff shown.

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉

“Accuracy is the twin brother of honesty; inaccuracy, of dishonesty.” — Nathaniel Hawthorne

This small but meaningful improvement brings training progress reporting closer to what users are actually processing, making logs clearer and more trustworthy across tasks. Thanks @Y-T-G for the thoughtful fix, and thanks as well to @onuralpszr and @UltralyticsAssistant for the collaboration. Your contribution helps make the Ultralytics training experience more intuitive for everyone.
